update mpt docs for permissioneddomains (#3251)
XRPLF/xrpl-dev-portal: docs/references/protocol/transactions/types/mptokenissuanceset.md

| Field               | JSON Type            | [Internal Type][] | Required? | Description |
|:--------------------|:---------------------|:------------------|:----------|-------------|

> | `DomainID`          | String - [Hash][]    | UInt256           | No        | The ledger entry ID of a permissioned domain that will manage access to the MPT. An empy value or `0` will remove permissioned domain access management. Any accounts that lose access to the MPT because of `DomainID` updates lose the ability to send or receive MPTs they already hold. The `tfMPTRequireAuth` flag must have been set in the `MPTokenIssuanceCreate` transaction to use permissioned domains. _(Requires the [PermissionedDomains amendment][] and [SingleAssetVault amendment][])_ |

Reviewer: `receive MPTs they already hold` I suggest adding something along the lines "unless they have been explicitly authorized by the MPT issuer"

> | `DomainID`          | String - [Hash][]    | UInt256           | No        | The ledger entry ID of a permissioned domain that will manage access to the MPT. An empy value or `0` will remove permissioned domain access management. Any accounts that lose access to the MPT because of `DomainID` updates--unless explicitly authorized by the MPT issuer--lose the ability to send or receive MPTs they already hold. The `tfMPTRequireAuth` flag must have been set in the `MPTokenIssuanceCreate` transaction to use permissioned domains.<br>{% amendment-disclaimer name="PermissionedDomains" /%}<br>{% amendment-disclaimer name="SingleAssetVault" /%} |

Reviewer 2: Same note as on create.

| Field               | JSON Type            | [Internal Type][] | Required? | Description |
|:--------------------|:---------------------|:------------------|:----------|-------------|

> | `DomainID`          | String - [Hash][]    | UInt256           | No        | The ledger entry ID of a permissioned domain that grants access to the MPT. An empy value or `0` will remove permissioned domain access management. Any accounts that lose access to the MPT because of `DomainID` updates--unless explicitly authorized by the MPT issuer--lose the ability to send or receive MPTs they already hold. The `tfMPTRequireAuth` flag must have been set in the `MPTokenIssuanceCreate` transaction to use permissioned domains.<br>{% amendment-disclaimer name="PermissionedDomains" /%}<br>{% amendment-disclaimer name="SingleAssetVault" /%} |

Reviewer 2: 1. Typo ("empy")
2. Avoid future tense.
3. Use an actual emdash instead of two hyphens. (Here's one you can copy-paste: — ) Or better yet, avoid phrasing sentences with this type of interjection in the first place.
4. The description feels a little confusing about when you lose the ability to send or receive MPTs.
5. The phrase "permissioned domain access management" feels a little ambiguous or confusing, like it's controlling who has access to manage the domain itself, and I don't think it's defined anywhere in those exact terms. Either rephrase into simpler language or shorten this description and link to a better description on a concept page.
6. Rather than talk about the exact transaction flag from the create transaction, talk about the current state of the MPT issuance (among other reasons, maybe that flag is mutable).

```suggestion
| `DomainID`          | String - [Hash][]    | UInt256           | No        | The ledger entry ID of a permissioned domain that grants access to the MPT. An empty value or `0` removes the permissioned domain from the MPT issuance so that only users who are explicitly approved by the issuer can send and receive the MPT. You can only set a `DomainID` if the MPT issuance has [**Require Auth**](/docs/concepts/tokens/fungible-tokens/multi-purpose-tokens#transferability-controls) enabled.) {% amendment-disclaimer name="PermissionedDomains" /%} {% amendment-disclaimer name="SingleAssetVault" /%} |
```

Reviewer 2: Note, I opened #3392 since it feels like we need better documentation on how to create and use allow-listing with MPTs (concept/tutorial/code sample/etc.).

| `tfMPTLock`        | `0x00000001` | 1             | Enable to lock balances of this MPT issuance. |
| `tfMPTUnlock`      | `0x00000002` | 2             | Enable to unlock balances of this MPT issuance. |

## Error Cases

Besides errors that can occur for all transactions, {% $frontmatter.seo.title %} transactions can result in the following [transaction result codes](../transaction-results/index.md):

| Error Code                | Description |
|:--------------------------|:------------|
| `tecNO_PERMISSION`        | You are trying to lock or unlock an MPT, but the `tfMPTCanLock` flag wasn't set in the `MPTokenIssuanceCreate` transaction. | 
| `temDISABLED`             | Either the `PermissionedDomains` or `SingleAssetVault` amendment is disabled. |

> | `temMALFORMED`            | You specified a `DomainID` and `Holder` value. You can only set one or the other in a single transaction. You can also receive this error if the transaction isn't changing anything; the transaction must either update a flag or modify the `DomainID`. |

Reviewer 2: `tecOBJECT_NOT_FOUND` and `tecNO_DST` are examples of a couple other type-specific errors that can occur with this transaction, based on a quick skim of [the source](https://github.com/XRPLF/rippled/blob/afc05659edbc22d04b13402f157832778556f7b7/src/xrpld/app/tx/detail/MPTokenIssuanceSet.cpp#L138).

| `tfMPTLock`        | `0x00000001` | 1             | Enable to lock balances of this MPT issuance. |
| `tfMPTUnlock`      | `0x00000002` | 2             | Enable to unlock balances of this MPT issuance. |

## Error Cases

Besides errors that can occur for all transactions, {% $frontmatter.seo.title %} transactions can result in the following [transaction result codes](../transaction-results/index.md):

| Error Code                | Description |
|:--------------------------|:------------|
| `temDISABLED`             | The `MPTokensV1` amendment is disabled. You will also receive this error if you include a `DomainID` field in the transaction, but the `PermissionedDomains` and `SingleAssetVault` amendments are both disabled. |

> | `tecNO_DELEGATE_PERMISSION` | You are attempting to use a delegate account with insufficient permissions to submit this transaction. |

Reviewer 2: I think this code is not specific to MPTokenIssuanceSet, and it's been replaced with a `ter` code in the next release also (relating to the bug that is the reason for `PermissionDelegationV1_1`)

Author: Good catch, removing this line.

| `tecNO_DST`               | The account specified in the `Holder` field doesn't exist. |

> | `tecNO_PERMISSION`        | - The `lsfMPTCanLock` flag isn't enabled, but you are attempting to lock or unlock an MPT.<br>- The `SingleAssetVault` amendment is disabled and you're trying to modify a `DomainID` field. | 

Reviewer 2: Stray bullet point?

```suggestion
| `tecNO_PERMISSION`        | The `lsfMPTCanLock` flag isn't enabled, but you are attempting to lock or unlock an MPT.<br>- The `SingleAssetVault` amendment is disabled and you're trying to modify a `DomainID` field. | 
```

Author: I've got 2 bullet points in this description.

> | `temMALFORMED`            | - You specified a `DomainID` and `Holder` value--only one can be set in a single transaction.<br>- You specified the same account for both `Acount` and `Holder`.<br>- The transaction isn't changing anything; it must either update a flag or modify the `DomainID`. |

Reviewer 2: `temMALFORMED` is a code that can occur for any transaction type.

```suggestion
```

Author: Same comment in issuancecreate. I think it's helpful to explain how you'd malform this specific transaction.
